convert/gemma.go

```diff
@@ -0,0 +1,140 @@
+package convert
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"strings"
+
+	"github.com/d4l3k/go-bfloat16"
+	"github.com/pdevine/tensor"
+	"github.com/pdevine/tensor/native"
+
+	"github.com/ollama/ollama/llm"
+)
+
+type GemmaModel struct {
```

Comment:
This shares the same struct members as `Mistral`. Would it be worth us either combining having a common struct, or combining the two in another way? This suggestion might be overkill if model architectures vary a lot

Reply:
I changed this and made a common `ModelData` struct which can be shared.